apps/src/code-studio/projectRedux.js

```diff
@@ -22,7 +23,11 @@ const initialState = {
   projectUpdatedAt: undefined,
   projectName: '',
   projectNameFailure: undefined,
-  showTryAgainDialog: false
+  showTryAgainDialog: false,
+  showWorkspaceAlert: false,
+  workspaceAlertType: undefined, // 'error' or 'warning'
```

Comment:
To enforce this contract a little better, you can create an enum-like mapping (similar to projectUpdatedStatuses) and then have callers reference that object instead of passing strings directly. For example if you had
```
workspaceAlertTypes: {
error: 'error',
...
}
```

Then when the action is dispatched
```
displayWorkspaceAlert(workspaceAlertTypes.error, ...);
```

Reply:
Great suggestion - implemented and also added notification as a third type of  alert type (refer to line 3107 in StudioApp.js)